utils/input_validators.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020 IOTA Stiftung
+ * https://github.com/iotaledger/iota_common
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "utils/input_validators.h"
+#include "common/model/transaction.h"
+
+bool is_tryte_with_trailing_zero(tryte_t const tryte) {
+  if (tryte == '9' || (tryte >= 'A' && tryte <= 'D') || (tryte >= 'W' && tryte <= 'Z')) {
+    return true;
+  }
+  return false;
+}
+
+bool is_address(tryte_t const *const address) {
+  if (strlen((char *)address) != NUM_TRYTES_ADDRESS) {
+    return false;
+  }
+  if (is_trytes(address, NUM_TRYTES_ADDRESS)) {
+    return is_tryte_with_trailing_zero(address[NUM_TRYTES_ADDRESS - 1]);
+  }
+  return false;
+}
+
+bool is_address_with_checksum(tryte_t const *const address) {
+  if (strlen((char *)address) != NUM_TRYTES_ADDRESS + 9) {
+    return false;
+  }
+  return is_tryte_with_trailing_zero(address[NUM_TRYTES_ADDRESS - 1]);
```

Comment:
Just double-checking, is it on purpose that `is_trytes` is not called on this function ? Do we assume we already called `is_address` before ?

Reply:
No, is_trytes should be called here.